UnityProject/Assets/Scripts/Machines/Jukebox.cs

```diff
@@ -67,33 +69,39 @@ public class Jukebox : NetworkBehaviour, IAPCPowered
 	private int currentSongTrackIndex = 0;
 	private float startPlayTime;
 
+	private string soundSpawnToken;
+
+	private async Task<AddressableAudioSource> GetAddressableAudioSourceFromCache()
+	{
+		AddressableAudioSource addressableAudioSourceParam = new AddressableAudioSource(SoundManager.Instance.MusicLibrary.ElementAt(currentSongTrackIndex));
+		AddressableAudioSource addressableAudioSourceFromCache = await SoundManager.GetAddressableAudioSourceFromCache(new List<AddressableAudioSource> { addressableAudioSourceParam });
+		return addressableAudioSourceFromCache;
+	}
+
 	public bool IsPlaying { get; set; } = false;
 
 	public string TrackPosition
 	{
 		get
 		{
-			return $"{currentSongTrackIndex + 1} / {musics.Count}";
+			return $"{currentSongTrackIndex + 1} / {SoundManager.Instance.MusicLibrary.Count}";
 		}
 	}
 
-	public string SongName
+	public async Task<string> GetSongNameAsync()
 	{
-		get
-		{
-			string songName = musics[currentSongTrackIndex].clip.name;
-			return $"{songName.Split('_')[0]}";
-		}
+		AddressableAudioSource addressableAudioSourceFromCache = await GetAddressableAudioSourceFromCache();
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)